qirlib/src/generation/qir/instructions.rs

```diff
@@ -157,11 +157,33 @@ pub(crate) fn emit<'ctx>(
             let qubit = get_value(generator, env, &inst.qubit);
             generator.emit_void_call(generator.qis_z_body(), &[qubit]);
         }
+        Instruction::BinaryOp(op) => emit_binary_op(generator, env, op),
         Instruction::Call(call) => emit_call(generator, env, call),
         Instruction::If(if_) => emit_if(generator, env, entry_point, if_),
     }
 }
 
+fn emit_binary_op<'ctx>(
+    generator: &CodeGenerator<'ctx>,
+    env: &mut Environment<'ctx>,
+    op: &BinaryOp,
+) {
+    let lhs = get_value(generator, env, &op.lhs).into_int_value();
+    let rhs = get_value(generator, env, &op.rhs).into_int_value();
+    let result = match op.kind {
+        BinaryKind::And => generator.builder.build_and(lhs, rhs, ""),
+        BinaryKind::Or => generator.builder.build_or(lhs, rhs, ""),
+        BinaryKind::Xor => generator.builder.build_xor(lhs, rhs, ""),
+        BinaryKind::Add => generator.builder.build_int_add(lhs, rhs, ""),
+        BinaryKind::Sub => generator.builder.build_int_sub(lhs, rhs, ""),
+        BinaryKind::Mul => generator.builder.build_int_mul(lhs, rhs, ""),
```

Comment:
@samarsha Do we have plan for dealing with this when we have float/fixed support? Calling `.into_int_value()` won't work for those ops in the future.

Reply:
Floating-point operators have different names (fadd, fsub, fmul, etc.). These operators will always be for integers only.

Fixed-points are represented as integers so they can use integer add/sub. Fixed-point mul and div intrinsics also have different names in LLVM.